README.md

```diff
@@ -131,9 +131,142 @@ You can add your customized Python code to process your input and output data. T
+        })
+
+    raise ValueError('{{"error": "unsupported content type {}"}}'.format(
+        context.request_content_type or "unknown"))
+
+
+def _process_output(data, context):
+    if data.status_code != 200:
+        raise ValueError(data.content.decode('utf-8'))
+
+    response_content_type = context.accept_header
+    prediction = data.content
+    return prediction, response_content_type
+```
+    
 You can also bring in external dependencies to help with your data processing. There are 2 ways to do this:
-1. If your model archive contains `code/requirements.txt`, the container will install the python dependencies at runtime using `pip install -r`.
-2. If you have pre-installed modules under `code/lib` directory in your model archive, the container will add the modules to the Python path. Note that if both `code/lib` and `code/requirements.txt` are present in the model archive, the `requirements.txt` will be ignored.
+1. If your model archive contains `code/requirements.txt`, the container will install the Python dependencies at runtime using `pip install -r`.
+2. If you are working in a network-isolation situation or if you don't want to install dependencies at runtime everytime your Endpoint starts or Batch Transform job runs, you may want to put pre-downloaded dependencies under `code/lib` directory in your model archive, the container will then add the modules to the Python path. Note that if both `code/lib` and `code/requirements.txt` are present in the model archive, the `requirements.txt` will be ignored.
+
+Your untarred model directory structure may look like this if you are using `requirements.txt`:
+
+        model1
+            |__[model_version_number]
+                |__variables
+                |__saved_model.pb
+        model2
+            |__[model_version_number]
+                |__assets
+                |__variables
+                |__saved_model.pb
+        code
+            |__inference.py
+            |__requirements.txt
+
+Your untarred model directory structure may look like this if you have downloaded modules under `code/lib`:
+
+        model1
+            |__[model_version_number]
+                |__variables
+                |__saved_model.pb
+        model2
+            |__[model_version_number]
+                |__assets
+                |__variables
+                |__saved_model.pb
+        code
+            |__lib
+                |__external_module
+            |__inference.py
+
+If you are not using SageMaker Python SDK, you should package the contents in model directory (including models, inference.py and external modules) in .tar.gz format and upload it to S3. [Here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/fw_utils.py#L146) is how we tar and upload the source directory to a S3 bucket. You can have other ways to do this but please make sure that the SageMaker Model entity has to point the compressed tar file in S3 as its model data URL.
```

Comment:
I've edited and expanded this -- could you review and edit the following, and include this?

### Packaging SageMaker Models for TensorFlow Serving

If you are not using SageMaker Python SDK, you should package the contents in model directory (including models, inference.py and external modules) in .tar.gz format in a file named "model.tar.gz" and upload it to S3. If you're on a Unix-based operating system, you can create a "model.tar.gz" using the `tar` utility:

```
tar -czvf model.tar.gz 12345 code
```

where "12345" is your TensorFlow serving model version which contains your SavedModel.

After uploading your `model.tar.gz` to an S3 URI, such as `s3://your-bucket/your-models/model.tar.gz`, create a [SageMaker Model](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateModel.html) which will be used to generate inferences. Set `PrimaryContainer.ModelDataUrl` to the S3 URI where you uploaded the `model.tar.gz`, and set `PrimaryContainer.Image` to an image following this format:

```
520713654638.dkr.ecr.{REGION}.amazonaws.com/sagemaker-tensorflow-serving:{TENSORFLOW_SERVING_VERSION}-{cpu|gpu}
```

Where `REGION` is your AWS region, such as "us-east-1" or "eu-west-1"; `TENSORFLOW_SERVING_VERSION` is one of the supported versions: "1.11" or "1.12"; and "gpu" for use on GPU-based instance types like ml.p3.2xlarge, or "cpu" for use on CPU-based instances like `ml.c5.xlarge`.


After creating your SageMaker Model, you can use it to create [SageMaker Batch Transform Jobs](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-batch.html)
 for offline inference, or create [SageMaker Endpoints](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-hosting.html) for real-time inference.

Reply:
done.